### ISSUE TITLE
No hamburger burg menu :(

### ISSUE THREAD

**Issue by L-a-n-g-o-l-i-e-r-s** [Noitidart/Profilist#39]
Hey there,

If I could contribute I would, just wondering how things are progressing, I tried building the latest extension and there isn't any burg menu. Was wondering if it's still a priority since it's a bit of a deal breaker.

Thanks for your hardwork!

**Comment by Noitidart** [comment]
I really appreciate your thoughts on trying to contribute. But you testing my addon and providing feedback with this topic is the best contribution you can give! Thank you! :)

What version are you using? The latest beta I uplaoded - https://addons.mozilla.org/en-US/firefox/addon/profilist/versions/beta - is 3.0b4 - it has a toolbar button there, when you click it, it opens the tab with the menu.

Firefox is moving to all HTML, so I created this new version in HTML. The HTML future is at least a year away though. So in v3.1 I plan to add the funcitonality to this hamburger menu.

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
3.05b, no burg and no toolbar button in edit or otherwise.

**Comment by Noitidart** [comment]
Whoa thats very weird.

When you click the hambruger menu you don't see the toolbar as seen in my screenshot:

![](http://i.imgur.com/qLPrlrV.png)

Clicking that, will open up a pinned tab to `about:profilist?html`. If you dont see that toolbar, can you please paste this URL into your url bar and see if it works.

Also what version of Firefox are you using? Nightly?

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:45.0) Gecko/20100101 Firefox/45.0

No I don't see that toolbar, but perhaps another extension is causing problems. I do have an extension that adds a restart button in the burg menu near the question mark.

**Comment by Noitidart** [comment]
I love that restart addon I use it too :)

That addon for sure shouldn't mess it up.

That's very interesting. I'm using Fx46 and it works fine.

Can you try navigating to `about:profilist?html` please

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
The address isn't valid

The URL is not valid and cannot be loaded. :

**Comment by Noitidart** [comment]
Ah that's a big problem. Can you please try to install the xpi from here:
https://github.com/Noitidart/Profilist/tree/bug1249929

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
Tried removing, restarting and reinstalling from that twice and still no dice.

**Comment by Noitidart** [comment]
Aw man thanks very much for your testing, I'll look into this closer.

To install the xpi, were you going to about:debugging and then clicking "Load Temporary Addon"?

![](http://i.imgur.com/eZTM0x4.png)

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
No, I have no issues with unsigned extensions and installation. I am using two right now unsigned that I compiled working fine on 45.

**Comment by Noitidart** [comment]
Ah ok thanks for that, I'll start digging at this. Thanks very much for this feedback!

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
I just tested via your method and it's still not displaying. I'll try disabling all my extensions and loading this one.

**Comment by Noitidart** [comment]
Can you install this addon to enable developer preferences - https://addons.mozilla.org/en-US/developers/addon/devprefs/ - and then install profilist and then paste here what is dumped to the browser console please?

Please install the one from that bug repo.

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
I don't know why but that link wants me to log in, not going to happen. I'm not going to sync my data with mozilla sorry. Did you link me to the right extension? which bug repo? the one from earlier?

**Comment by Noitidart** [comment]
This repo here please - https://github.com/Noitidart/Profilist/tree/bug1249929
There is Profilist.xpi in that repo. Please try that one.

**Comment by Noitidart** [comment]
Ohhh the dev prefs addon is here - https://addons.mozilla.org/en-US/firefox/addon/devprefs/ - yeah i posted wrong link haha shoot

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
http://pastebin.com/h6qH5UFv

**Comment by Noitidart** [comment]
Oh very interesting, it's getting stuck at step0.

Was there anything after that `worker: step0` line?

You should have seen something starting with `msgFrom_fsReturnIconset_onPageReady`

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
http://pastebin.com/jZ5x32E2

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
http://pastebin.com/taXGWBcJ clicked dropped down arrows for more detail.

**Comment by Noitidart** [comment]
That's real interesting. I don't know why its getting stuck at step0 I'll dig into this.

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
Yes, if I recall for some reason any version that I installed after a specific 2.0 version wouldn't work at all for my FF.

**Comment by Noitidart** [comment]
Ohhh you had 2.0 installed!! Ok lets do some resetting.

Please go to the the "run dialog" or hit Win Key + R

Then type in %APPDATA%\Mozilla\Firefox

Then delete the profilist_data folder.

Then open firefox from the regular profile manager. By going to "run dialog" again, and then type in `firefox -P` then exit. That will reset your profiles.ini - then launch up firefox the way you normally do please.

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
I had it installed a long time ago but have since made a new profile. I'll see if that file is there.

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
This is still an issue in v4 beta when built with xpiler, I cannot access a burg menu, nor can I access it via about:profilist or about:profilist?html. "The address isn’t valid" when even trying to click options via the about:addons options. 

New or old profile makes no difference. 

Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:45.0) Gecko/20100101 Firefox/45.0
(49.01 x64 build now).

**Comment by Noitidart** [comment]
Hey @L-a-n-g-o-l-i-e-r-s ! Thanks for trying out v4, v4 is a total redo and it's going to show in the burger menu. Currently v4 is not hooked up to the profiles.ini file.

There is a big discussion on webextension future which will block all my addons. I'm trying to figure out how I can do this in a future proof work.

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
" Parker Lewis September 4, 2016 at 5:35 pm #

@CHEF-KOCH
> "Mozilla needs a:
> * Concept
> * A better engine and a modern coding language
> * More options like all other browsers to make it 'unique' again,
> do not copy features from Chrome! "

Mozilla has all those things, check it out for yourself. People just repeat random drama stuff they read and spread negativity, don't let yourself be affected, look up cold technical facts instead.

The main thing that holds Mozilla back is add-ons, and that is going to be solved with practically no sacrifice with WebExtensions + native.js. Firefox add-ons can't be ditched, they're too great. Once they have moved to WebExtensions, all the compatibility-breaking tech can be unleashed without damage to the ecosystem.

That includes everything you suggest.

    Yes, an API is a serious limitation over total freedom. But like I said native.js will be able to do anything that old-style add-ons can do.

    From Bill McCloskey, who is in charge of this at Mozilla:
    "I don't think there will be any technical limitations on what sort of XUL or XPCOM access is permitted from native.js."

    The 'limitation' on native.js only concerns ease of development. Instead of minding your own business as a developer, doing your own thing and publishing your add-on, you will have to build a native.js module that will be subjected to more careful review than regular add-ons. And Mozilla and other add-on developers will be looking at your native.js module in case it breaks, so even if you die or disappear they can fix it.

    If it was like you said, indeed this would be a tradeoff. But it's not, we're only trading some inconvenience versus the good things talked about here and there in this article's comments."

I chopped it up a bit, it was some comments on ghacks, honestly, it made me a lot less angry and it seemed logical, I don't know what happened over the weekend, but this seems logical.

**Comment by Noitidart** [comment]
Thanks for sharing that info! What caused the situation was I think native.js implementation landed and it was nothing like what was proposed - https://wiki.mozilla.org/WebExtensions/Experiments#Other_proposals

:( may you email me at noitidart@gmail.com and I'll forward you the email chain of the discussion.

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
That is very disappointing to hear!

**Comment by Noitidart** [comment]
I started v4 last week, and now I have to start v5 which is a webextension. No burger menu possibility :'(

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
Can we at least drag an icon into the burger menu?

**Comment by Noitidart** [comment]
Yes it will be an icon in the toolbar, clicking it will open a new tab. I can't put the menu inside a panel as resizing the height of the menu dynamically really messes with the panel. :(
